### PR TITLE
feat: update title on note content change

### PR DESCRIPTION
**Pull request by anig1scur**
So that you could get one note from dock by title directly

image

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -1,5 +1,5 @@
 project('sticky-notes',
-          version: '0.2.1',
+          version: '0.2.2',
```
Comment (vixalien) [inline meson.build:2]: please don't bump the version here.

**Review by vixalien — COMMENTED** [review]
A better approach would be to add another property named `title` to the `Note` (it's in util.ts).

Then you can easily bind the note's title to the note's content, by doing something similar to this in the constructor of the Note:

```ts
this.bind_property_full("title", this, "content", flags, (_, content) => {
 // return the first n-chars of the content, or the first line
  }, null);
```

Then you can create yet another binding from the note's title to the note window's title

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -32,11 +32,10 @@ jobs:
         run: |
           flatpak --system remote-add --if-not-exists flathub-beta https://flathub.org/beta-repo/flathub-beta.flatpakrepo
           flatpak --system install -y --noninteractive flathub-beta org.freedesktop.Sdk.Extension.node20/${{ matrix.arch }}/24.08beta
+
       - uses: flatpak/flatpak-github-actions/flatpak-builder@v6
         with:
           bundle: com.vixalien.sticky.flatpak
```
Comment (vixalien) [inline .github/workflows/build.yml:38]: Thanks, but these changes belong in another PR

File: src/util.ts [hunk 3]
```diff
@@ -211,13 +214,31 @@ export class Note extends GObject.Object {
...
+
+    // @ts-expect-error incorrect types
+    this.bind_property_full(
+      "content",
+      this,
+      "title",
+      GObject.BindingFlags.SYNC_CREATE,
+      (_, content) => {
+        if (!content) return [false, ""];
+        return [true, content.substring(0, 24)]
```
Comment (vixalien) [inline src/util.ts:226]: You can do this so that it returns the first line if it's there.

```suggestion
        return [true, content.split("\n")[0].substring(0, 24)]
```

Also this should add an ellipsis (…) if the content has been truncated.

File: src/window.ts [hunk 4]
```diff
@@ -98,6 +98,12 @@ export class Window extends Adw.ApplicationWindow {
     this.default_width = note.width;
     this.default_height = note.height;
 
+    this.update_title();
+
+    this.note.connect("title-changed", () => {
```
Comment (vixalien) [inline src/window.ts:103]: You can also use a binding here, to bind the note's title to the window's title directly

**Review by vixalien — COMMENTED** [review]
Thank you!!

File: src/window.ts [hunk 5]
```diff
@@ -98,6 +98,13 @@ export class Window extends Adw.ApplicationWindow {
     this.default_width = note.width;
     this.default_height = note.height;
 
+    this.note.bind_property(
+      "title",
+      this,
+      "title",
+      GObject.BindingFlags.SYNC_CREATE
```
Comment (vixalien) [inline src/window.ts:105]: I think a window having an empty title is problematic. Could you modify the binding so that it sets the title to `_("Sticky Note")` if it's empty?

Comment (anig1scur) [reply src/window.ts]: I set the title to `Untitled Note` if it's empty

File: src/window.ts [hunk 6]
```diff
@@ -98,6 +98,20 @@ export class Window extends Adw.ApplicationWindow {
...
     this.default_height = note.height;
 
+    this.note.bind_property_full(
+      "title",
+      this,
+      "title",
+      GObject.BindingFlags.SYNC_CREATE,
+      (_, title) => {
+        if (!title) {
+          return [true, "Untitled Note"];
```
Comment (vixalien) [inline src/window.ts:108]: I feel like "Sticky Note" may be better here.

Also, wrap the string in a `_("string")` so that it may get picked up by gettext (the translation infrastructure).

```suggestion
          return [true, _("Sticky Note")];
```

Comment (anig1scur) [reply src/window.ts]: done !

**Comment by vixalien** [comment]
Thank you!

I still need to test this locally, but looks good

**Review by vixalien — APPROVED** [review]
